website/docs/terminology.md

```diff
@@ -75,10 +75,11 @@ See also: [Code reference for control root]
 
 ### Deterministic Builds
 
-Our [cargo risczero] tool offers a deterministic compilation from user-written Rust code to RISC-V binary.
-Compiled binary files are identified by their [imageID].
+A compilation process is called "deterministic" (or "reproducible") if it reliably produces the same binary file, on a byte-by-byte level.
 
-Deterministic builds are made possible by running the `cargo` compiler inside a Docker container. Without deterministic builds, the zkVM only proves correct execution of a given binary file. With deterministic builds, users can prove correct execution of Rust code.
+In the context of RISC Zero application development, deterministic builds are necessary to ensure a clear linkage between the [guest program] and the resulting [ImageID].
```

Comment:
I'd recommend clarifying this to be in reference to the source code, because I first read this as referring to the ELF, from which there is already a deterministic translation to an image ID. (Also, in prose, I feel like there should be a space in "Image ID")
```suggestion
In the context of RISC Zero application development, deterministic builds are necessary to ensure a clear linkage between the source code for [guest program] and the resulting [Image ID].
```

Reply:
```suggestion
In the context of RISC Zero application development, deterministic builds are necessary to ensure a clear linkage between the source code for the [guest program] and the resulting [Image ID][ImageID].
```